neo/Cryptography/ECC/ECDsa.cs

```diff
@@ -10,15 +10,25 @@ public class ECDsa
         private readonly byte[] privateKey;
         private readonly ECPoint publicKey;
         private readonly ECCurve curve;
+        private readonly BigInteger d;
 
         public ECDsa(byte[] privateKey, ECCurve curve)
             : this(curve.G * privateKey)
         {
+            if (privateKey == null)
+                throw new ArgumentNullException();
+            this.d = new BigInteger(privateKey.Reverse().Concat(new byte[1]).ToArray());
+            if (d == 0 || d >= curve.N)
+                throw new ArgumentException();
+
             this.privateKey = privateKey;
         }
```

Comment:
From the code, `d` is the private key, so there is no need to save both `d` and the `privateKey`. By the way, these modifications are meaningless. Beyond this constructor, almost all classes in the ECC namespace are not actually used.

Reply:
Thanks dalao for the suggestion. We have removed the `privateKey ` property. 
As the code is there, we want to make sure it is correct.  Just as you said, most classes in ECC namespace are not used, we plan to remove them in the next PR for Neo3.